### ISSUE TITLE
Restore result summary subhead

### ISSUE THREAD

**Issue by mmiddaugh** [department-of-veterans-affairs/va.gov-team#7555]
## User Story or Problem Statement
As a Veteran using a screenreader, I need to be able to differentiate the search results list from the other page content. 

## Background
This functionality was previously implemented but was [hidden](https://github.com/department-of-veterans-affairs/va.gov-team/issues/6775) when it was determined that the number of results was incorrect and misleading to Veterans. 

![image](https://user-images.githubusercontent.com/55411834/87810450-41640400-c81a-11ea-9994-dfd7b29b3fbf.png)

## Current functionality
Reading order goes to the first of the search results, without any introduction to provide context.
After entering search criteria and the Search button, focus remains on the Search button.
**Existing functionality currently has several 508-defect-1 issues logged against it..** 
- [#7240](https://github.com/department-of-veterans-affairs/va.gov-team/issues/7240) 
- [#7239](https://github.com/department-of-veterans-affairs/va.gov-team/issues/7239)
- [#7243](https://github.com/department-of-veterans-affairs/va.gov-team/issues/7243)

## Goal
We need a heading Introducing the search results, without the X–Y of Z.  Adding a heading will give context to Veterans who use screenreaders and generally improve the user experience for all Veterans. 

**NOTE: A separate issue will be created to include the number of results displayed. This is currently blocked by the PPMS data structure and may _or may not_ be possible to implement in the future.** 

---

## Tasks
- [ ] Add a heading (h2) "Search Results" which appears above the list of results and includes user search criteria 

> **Search results** for <**VA facility type**> near **"location parameters"**

- [ ] Ensure focus goes to results summary subhead consistent with user flow
- [ ] Do not update the result summary until the search is executed.
- [ ] Change search result facility names to h3

## Acceptance Criteria
For all Facility Locator searches:
- [x] Results summary subhead displays when search results are returned. 
- [x] Results summary subhead does not prematurely load
- [x] When changing criteria after a previous search, the result summary is not updated until the new search is executed
- [x] Heading levels do not skip
- [x] Focus moves from the Search button to this heading, as indicated in ticket #7240
- [ ] Heading includes user search criteria (**Search results** for <**VA facility type**> near **"location parameters"**)
- [ ] Heading updates with new search criteria if new search is initiated
- [ ] No heading is seen while a search is in progress
- [ ] Only the standard error displays if no search results are found (no heading displays)
- [ ] Veterans using screenreaders understand the context of the search results list
- [ ] Responsive design: all elements are fully visible
- [ ] Keyboard test confirms focus
- [ ] New functionality is covered by E2E test

---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `tools-be`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

**Comment by mmiddaugh** [comment]
@lildziner please weigh in on the display for the subhead: 
**Search results** for <**VA facility type**> near **"location parameters"**

**Comment by mmiddaugh** [comment]
@mpelzsherman, the following notes pertain to a recent conversation with Jennifer (508/Assessibility):
- Results summary s/b h2
- Search results s/b h3

**Comment by lildziner** [comment]
@mpelzsherman Regarding styling decisions for this ticket:
- Font size: "base" (16 px) 
- Font family: Source Sans Pro
- Search parameters get bolded and should be in quotations. 

![search-results-summary](https://user-images.githubusercontent.com/4960080/88585493-29ae2c00-d021-11ea-9b5e-e6fccbb19c72.png)

**Comment by mmiddaugh** [comment]
@jenstrickland this is ready for your validation.

**Comment by jenstrickland** [comment]
Validated. Thank you!

**Comment by mpelzsherman** [comment]
Something I noticed: when a search returns zero results, we zoom out and re-try the search. However, the focus is still moved to the search results header (which is empty in that case) when the empty results are returned.

Fixing this will require a new ticket.

**Comment by jenstrickland** [comment]
Focus should only be moved the results header when there are search results. Otherwise, it should be moved to a message, no results found.

Would it be possible to have a structure like this:

Your search results [- for "VA health" near  "Arlington, Virginia 22202"]

**Comment by mpelzsherman** [comment]
We could do that. There are different use cases here though: no results found **ever**, even after zooming out, vs. no results found in the initial search. In any case, we should not set focus on the h2 unless/until there are search results. Can we write up a bug for this, @mmiddaugh ?
